### DISCUSSION TITLE
Vercel Deploy: `RangeError: Maximum call stack size exceeded`

### DISCUSSION THREAD

**Discussion by AlexJSully** [vercel/community#5449] [Help] [answered]
### Summary

Hi all!

I'm trying to push my changes I made to my code (open source at https://github.com/AlexJSully/AlexJSully-Portfolio & https://alexjsully.me/) and during deploy to preview build I am getting the following error during Vercel Next build. 

When I run `npm run build` locally, I get no issues but on Vercel deploy CI/CD, I get it.

How do I fix or address this?

Error:
```
[17:23:49.978]  ✓ Compiled successfully
[17:23:49.979]    Linting and checking validity of types ...
[17:24:00.091]    Collecting page data ...
[17:24:01.948]    Generating static pages (0/7) ...
[17:24:01.985] 
   Generating static pages (1/7) 
[17:24:02.142] 
   Generating static pages (3/7) 
[17:24:02.498] 
   Generating static pages (5/7) 
[17:24:02.802] 
 ✓ Generating static pages (7/7) 
[17:24:03.158]    Finalizing page optimization ...
[17:24:03.158]    Collecting build traces ...
[17:24:12.983] RangeError: Maximum call stack size exceeded
[17:24:12.983]     at parse (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:6313)
[17:24:12.983]     at picomatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:21670)
[17:24:12.983]     at picomatch (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:19637)
```
![Untitled](https://github.com/vercel/community/assets/2214284/49534770-6e1f-4c18-bc30-1ef9c5799103)

#### Relevant packages installed:

dep:
- "@sentry/integrations": "^7.93.0",
- "@sentry/nextjs": "^7.93.0",
- "@vercel/speed-insights": "^1.0.3",
- "firebase": "^10.7.1",
- "next": "^14.0.4",
- "next-pwa": "^5.6.0",
- "react": "^18.2.0",
- "react-dom": "^18.2.0",
- "sharp": "^0.33.2",
- "webpack": "^5.89.0"

devDep:
- "@svgr/webpack": "^8.1.0",
- "@types/next-pwa": "^5.6.9",
- "@types/node": "^20.11.0",
- "@types/react": "^18.2.47",
- "@types/react-dom": "^18.2.18",
- "cypress": "^13.6.2",
- "eslint": "^8.56.0",
- "eslint-config-next": "^14.0.4",
- "eslint-config-prettier": "^9.1.0",
- "eslint-plugin-cypress": "^2.15.1",
- "eslint-plugin-jsx-a11y": "^6.8.0",
- "eslint-plugin-prettier": "^5.1.3",
- "eslint-plugin-react": "^7.33.2",
- "eslint-plugin-react-hooks": "^4.6.0",
- "prettier": "^3.2.1",
- "sass": "^1.69.7",
- "typescript": "^5.3.3"

### Steps to Reproduce

- Push to GitHub dev channel
- GitHub Actions CI/CD run Vercel deploy
- Vercel deploy fails on `npm run build`

**Comment by njwright92** [comment]
I'm having the same issue too was working fine few days ago, tested deploying it with netlify and it worked, so I don't know what's going on with Vercel right now

**Comment by swarnava** [comment]
Hi,

We are investigating this issue.

**Reply by njwright92** [reply]
Hope so worked fine few days ago updated couple href link addressses now I'm exceeding call
Stack????

**Reply by swarnava** [reply]
Can you check if you are using `sharp` module?

**Reply by patricktree** [reply]
Hey, I submitted some information here, I think it helps! https://github.com/lovell/sharp/issues/3944#issuecomment-1890513450

**Reply by njwright92** [reply]
Removed the sharp package as I don't need it anymore and now it works thank you.

**Reply by juxuanu** [reply]
@swarnava please see https://github.com/lovell/sharp/issues/3944#issuecomment-1890587296

**Reply by swarnava** [reply]
Thanks for sharing! Yes we will investigate this today and let you know once we fix the issue.

**Comment by swarnava** [comment]
@AlexJSully it looks like you have upgraded to Sharp module recently which likely causing the issue. Can you downgrade "sharp": to "0.33.1" and let us know if you are still seeing this error?

**Reply by AlexJSully** [reply]
Sadly that did not fix it. Still getting the same issue on my Vercel deploy build.

The `npm run build` is working fine locally and on my code QA GitHub Action CI/CD: https://github.com/AlexJSully/AlexJSully-Portfolio/actions/runs/7512120969/job/20452461447 - just not on Vercel deploy build

**Reply by swarnava** [reply]
Can you pin 0.33.1 (without `^` )and give it a try? Please make sure you also remove package-lock.json which may latest version pinned. 

- I cloned your repo
- ran `npm i`
- ran `npm i sharp@0.33.1`
- ran `vercel build`

It seems worked for me

**Reply by AlexJSully** [reply]
That worked! Thank you greatly!

**Reply by StanLachie** [reply]
Hey guys, these fixes have fixed most of my issues but have caused one more big issue, I've spent literally all day trying to fix it with no luck.

Within my localhost, everything works perfectly. However when pushed to prod on Vercel, I run into this error on my API route: 405 (Method not found). I'm having a lot of trouble debugging and troubleshooting as once again, it works perfectly on localhost.

Here's the API route:

```js
import { NextRequest, NextResponse } from "next/server";

import { cookies } from "next/headers";
import { createServerClient } from "@supabase/ssr";
import { getUser } from "@/lib/utils";
import sharp from "sharp";

export async function POST(request: NextRequest) {
  // Check if environment variables are set
  const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
  const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
  if (!supabaseUrl || !supabaseAnonKey) {
    return NextResponse.json(
      { error: "Server configuration error." },
      { status: 500 }
    );
  }

  const cookieStore = cookies();
  const supabase = createServerClient(supabaseUrl, supabaseAnonKey, {
    cookies: {
      get(name: string) {
        return cookieStore.get(name)?.value;
      },
    },
  });

  try {
    const formData = await request.formData();
    const file = formData.get("file");
    const fileName = formData.get("fileName");

    if (!file || typeof file !== "object" || !(file instanceof Blob)) {
      return NextResponse.json(
        { error: "Invalid file provided." },
        { status: 400 }
      );
    }

    if (!fileName || typeof fileName !== "string") {
      return NextResponse.json(
        { error: "Invalid file name provided." },
        { status: 400 }
      );
    }

    const reformattedFileName = fileName
      .replace(/\s/g, "_")
      .replace(/[^\w.-]/g, "");

    const convertedImage = await sharp(Buffer.from(await file.arrayBuffer()))
      .resize(750, 750, { fit: "cover", position: "center" })
      .webp()
      .toBuffer();

    const user = await getUser(cookieStore);
    if (!user) {
      return NextResponse.json(
        { error: "User authentication failed." },
        { status: 401 }
      );
    }

    const { data, error: selectError } = await supabase
      .from("restaurants")
      .select("id")
      .eq("client", user.id)
      .single();

    if (selectError) {
      throw selectError;
    }

    const { error: uploadError } = await supabase.storage
      .from("f3ast-client-menu-assets")
      .upload(`${data.id}/${reformattedFileName}.webp`, convertedImage, {
        contentType: "image/webp",
      });

    if (uploadError) {
      throw uploadError;
    }

    return NextResponse.json({ Message: "Success", status: 201 });
  } catch (error: any) {
    console.error("Error occurred: ", error);
    return NextResponse.json({
      Message: "Failed",
      status: 500,
      error: error.message,
    });
  }
}
```

**Reply by StanLachie** [reply]
@swarnava

**Reply by swarnava** [reply]
@StanLachie  This looks like a separate issue. Please file a new ticket / discussion. Thanks!

**Comment by arnabonline** [comment]
Downgrading the version of sharp is a temporary fix. Is Vercel investigating this issue so that sharp can be upgraded?

**Reply by FleetAdmiralJakob** [reply]
I tried to downgrade to sharp 0.33.1 and didn't fix it.

**Reply by kc97ble** [reply]
@FleetAdmiralJakob Instead of downgrading `sharp`, try upgrading `next`:

```diff
-    "next": "14.0.3",
+    "next": "14.1.0",

-    "eslint-config-next": "14.0.3",
+    "eslint-config-next": "14.1.0",
```

The change above works for me.

I am using `"sharp": "^0.33.2"` (resolved to 0.33.2)

**Comment by swarnava** [comment, marked as answer]
Hi,

The fix is now available in `v14.1` of Next.js where you can use latest `sharp` package.

**Reply by krnrnee** [reply]
Worked for me, thanks!

**Reply by boris-szl** [reply]
Working well!!!

**Reply by OU9999** [reply]
thanks!! working well!!
